### PR TITLE
scheduler: enhance output when scheduler config update

### LINKED ISSUE #4607: No update logs when set balance-hot-region-scheduler attributes
### What did you do?
/var/lib/pd/log # grep 'config is updated' * |awk -F" " '{print $1,$2,$3,$4,$5,$6,$7,$8,$9}'
[2022/01/20 10:01:37.910 +00:00] [INFO] [server.go:957] ["PD server config is
[2022/01/20 11:10:56.768 +00:00] [INFO] [server.go:835] ["schedule config is updated"]
[2022/01/20 11:12:36.790 +00:00] [INFO] [server.go:835] ["schedule config is updated"]
[2022/01/20 11:14:16.774 +00:00] [INFO] [server.go:835] ["schedule config is updated"]
[2022/01/20 11:15:56.775 +00:00] [INFO] [server.go:835] ["schedule config is updated"]
[2022/01/20 11:17:36.783 +00:00] [INFO] [server.go:835] ["schedule config is updated"]
[2022/01/20 11:19:16.775 +00:00] [INFO] [server.go:835] ["schedule config is updated"]
[2022/01/20 11:20:56.783 +00:00] [INFO] [server.go:835] ["schedule config is updated"]
[2022/01/20 11:21:25.572 +00:00] [INFO] [server.go:835] ["schedule config is updated"]
[2022/01/21 05:44:51.526 +00:00] [INFO] [server.go:835] ["schedule config is updated"]
[2022/01/21 05:46:31.542 +00:00] [INFO] [server.go:835] ["schedule config is updated"]
[2022/01/21 05:48:11.541 +00:00] [INFO] [server.go:835] ["schedule config is updated"]
[2022/01/21 05:49:51.532 +00:00] [INFO] [server.go:835] ["schedule config is updated"]
[2022/01/21 05:51:31.540 +00:00] [INFO] [server.go:835] ["schedule config is updated"]
[2022/01/21 05:53:11.541 +00:00] [INFO] [server.go:835] ["schedule config is updated"]
[2022/01/21 05:54:05.620 +00:00] [INFO] [server.go:835] ["schedule config is updated"]

2022-01-21T13:59:06.490+0800    INFO    pd-basic/pdbasic_schedule_test.go:299   /pd-ctl scheduler config balance-hot-region-scheduler set strict-picking-store false:Success!
2022-01-21T13:59:06.490+0800    INFO    oltppkg/util.go:62      workload run    {"wait": "5m0s"}

### What did you expect to see?
See update logs when set balance-hot-region-scheduler parameters.

### What did you see instead?
No related logs print.

### What version of PD are you using (`pd-server -V`)?
/ # /pd-server -V
Release Version: v5.4.0-nightly
Edition: Community
Git Commit Hash: 964569d45ac0685110d12665e6ade8680042c8e5
Git Branch: heads/refs/tags/v5.4.0-nightly
UTC Build Time:  2022-01-20 06:11:07

Minio logs can be get from tcms via plan id: 600774, testbed: pd-schedule-hongm-tps-600774-1-763

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Close #4607

### What is changed and how does it work?

```commit-message
```
1. reduce unnecessary config check in pd-ctl.
2. return result when run successfully
3. add log when config changed

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
image

### Release note

```release-note
None.
```

### CODE REVIEW

File: server/api/scheduler_test.go [hunk 1]
```diff
@@ -170,7 +170,7 @@ func (suite *scheduleTestSuite) TestAPI() {
 				suite.NoError(err)
 				err = tu.CheckPostJSON(testDialClient, updateURL, body,
 					tu.Status(re, http.StatusBadRequest),
-					tu.StringEqual(re, "\"config item not found\"\n"))
+					tu.StringEqual(re, "\"Config item not found\"\n"))
```
Comment (lhy1024) [inline server/api/scheduler_test.go:173]: I will add and fix test after #6929 merged

File: tools/pd-ctl/pdctl/command/global.go [hunk 2]
```diff
@@ -207,7 +207,7 @@ func requestJSON(cmd *cobra.Command, method, prefix string, input map[string]int
 		cmd.Printf("Failed! %s\n", err)
 		return
 	}
-	cmd.Println("Success!")
+	cmd.Printf("Success! %s\n", msg)
```
Comment (nolouch) [inline tools/pd-ctl/pdctl/command/global.go:210]: It's better to trim the msg with `strings.Trim(msg, "\"")`

File: tools/pd-ctl/pdctl/command/scheduler.go [hunk 3]
```diff
@@ -714,32 +713,7 @@ func postSchedulerConfigCommandFunc(cmd *cobra.Command, schedulerName string, ar
...
-		if len(priorities) < 2 {
-			cmd.Println("priorities should have at least 2 dimensions")
-			return
-		}
-		input[key] = priorities
-		if len(priorities) != len(prioritiesMap) {
-			cmd.Println("priorities shouldn't be repeated")
-			return
-		}
+		input[key] = strings.Split(value, ",")
```
Comment (nolouch) [inline tools/pd-ctl/pdctl/command/scheduler.go:716]: why remove it? I think the check `priority should be one of [%s, %s, %s]` is necessary.

Comment (lhy1024) [reply tools/pd-ctl/pdctl/command/scheduler.go:716]: http interface will check it also

File: pkg/schedule/schedulers/balance_witness.go [hunk 4]
```diff
@@ -76,17 +76,18 @@ func (conf *balanceWitnessSchedulerConfig) Update(data []byte) (int, interface{}
 			return http.StatusBadRequest, "invalid batch size which should be an integer between 1 and 10"
 		}
 		conf.persistLocked()
-		return http.StatusOK, "success"
+		log.Info("balance-leader-scheduler config is updated", zap.ByteString("old", oldc), zap.ByteString("new", newc))
```
Comment (HuSharp) [inline pkg/schedule/schedulers/balance_witness.go:79]:
```suggestion
		log.Info("balance-witness-scheduler config is updated", zap.ByteString("old", oldc), zap.ByteString("new", newc))
```

File: pkg/schedule/schedulers/hot_region_config.go [hunk 5]
```diff
@@ -421,7 +421,9 @@ func (conf *hotRegionSchedulerConfig) handleSetConfig(w http.ResponseWriter, r *
 	newc, _ := json.Marshal(conf)
 	if !bytes.Equal(oldc, newc) {
 		conf.persistLocked()
-		rd.Text(w, http.StatusOK, "success")
+		log.Info("hot region scheduler config is updated", zap.String("old", string(oldc)), zap.String("new", string(newc)))
```
Comment (HuSharp) [inline pkg/schedule/schedulers/hot_region_config.go:424]:
```suggestion
		log.Info("hot-region-scheduler config is updated", zap.String("old", string(oldc)), zap.String("new", string(newc)))
```

File: tools/pd-ctl/pdctl/command/scheduler.go (hunk 3 above)
Comment (HuSharp) [reply tools/pd-ctl/pdctl/command/scheduler.go:716]: Can you point out the exact address which checked?

**Review by HuSharp — COMMENTED** [review]
btw, I noticed there are too many `api/v1/scheduler-config` var, do we need to unify them?

File: tools/pd-ctl/pdctl/command/scheduler.go (hunk 3 above)
Comment (lhy1024) [reply tools/pd-ctl/pdctl/command/scheduler.go:716]: https://github.com/tikv/pd/blob/39cff3bb2e5bea126212d92c90687ca63664d732/pkg/schedule/schedulers/hot_region_config.go#L352

**Review by nolouch — APPROVED** [review]
lgtm

**Review by HuSharp — APPROVED** [review]
LGTM!

**Pull request merged**